refactor(transformer/class-properties): remove pointless method

overlookmotel opened the pull request:
#7516 removed the need for a special method to handle `export default class {}`. Delete it.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7592?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7592** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7592?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7592?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#7516** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7516?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7589** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7589?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7588** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7588?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7587** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7587?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7586** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7586?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7532** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7532?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7531** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7531?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7533** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7533?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel commented:
@Dunqing Getting this working has had me tearing my hair out for days. I suggest not giving it a full review - I'm as sure as I can be that it's correct now, and it'll only have you tearing your hair out too!

But please do take a look at the changes to `transform_private_field_expression_impl` in #7532 and #7516, as I've had to change the APIs related to private fields, which will affect your work on `this?.#prop`. I don't think it'll make a dramatic difference, but your PR will need a few tweaks to align with the changes here. Happy to help if it causes you trouble.

Dunqing commented:
### Merge activity

* **Dec 3, 2:44 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(transformer/class-properties): remove pointless method (#7592)

#7516 removed the need for a special method to handle `export default class {}`. Delete it.
